servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -376,6 +377,12 @@ private Throwable enrichError(final Throwable t) {
         } else {
             if ((closeReason = this.closeReason) != null) {
                 throwable = wrapWithCloseReason(closeReason, t);
+                if (closeReason == CHANNEL_CLOSED_INBOUND && t instanceof StacklessClosedChannelException &&
+                        !NativeTransportUtils.isEpollAvailable() && !NativeTransportUtils.isKQueueAvailable()) {
```

Comment:
Instead of checking transports in general here, consider checking the `EventLoop`/`Channel` type and adding a utility method in `NativeTransportUtils` for this type of check (e.g. `channelNotifiesCloseIfIdel`). this would be similar to the approach in `isUnixDomainSocketSupported` and helps consolidate the "transport conditional logic" in a single place.